eq-author-api/schema/resolvers/importing.js

```diff
@@ -0,0 +1,88 @@
+const {
+  getPagesByIds,
+  getFolderById,
+  getSectionById,
+  getSectionByFolderId,
+  stripQCodes,
+  remapAllNestedIds,
+} = require("./utils");
+
+const createFolder = require("../../src/businessLogic/createFolder");
+
+const { getQuestionnaire } = require("../../db/datastore");
+const { UserInputError } = require("apollo-server-express");
+
+const { createMutation } = require("./createMutation");
+
+module.exports = {
+  Mutation: {
+    importQuestions: createMutation(
+      async (_, { input: { questionnaireId, questionIds, position } }, ctx) => {
+        const { sectionId, folderId, index: insertionIndex } = position;
+
+        if (!sectionId && !folderId) {
+          throw new UserInputError(
+            "Target folder or section ID must be provided."
+          );
+        }
+
+        const sourceQuestionnaire = await getQuestionnaire(questionnaireId);
+        if (!sourceQuestionnaire) {
+          throw new UserInputError(
+            `Questionnaire with ID ${questionnaireId} does not exist.`
+          );
+        }
+
+        const pages = getPagesByIds(
+          { questionnaire: sourceQuestionnaire },
+          questionIds
+        );
+        if (pages.length !== questionIds.length) {
+          throw new UserInputError(
+            `Not all page IDs in [${questionIds}] exist in source questionnaire ${questionnaireId}.`
+          );
+        }
+
+        // Re-create UUIDs, strip QCodes, routing and skip conditions from imported pages
+        // Keep piping intact for now - will show "[Deleted answer]" to users when piped ID not resolvable
```

Comment:
It should strip out piping; the frontend UI will inform the user that piping won't be imported. RAD agreed that we'll address piping in the next iteration.

If you think doing this will be more work than what you have now, can you bring it up with the RAD team and then update the ticket with the result, please? We'll then also need to update the design to change the guidance we give to users.